include/net/golioth/rpc.h

```diff
@@ -11,11 +11,12 @@
 #include <zephyr/net/coap.h>
 #include <zephyr/kernel.h>
 
-struct _QCBOREncodeContext;
-struct _QCBOREncodeContext;
-
-typedef struct _QCBORDecodeContext QCBORDecodeContext;
-typedef struct _QCBOREncodeContext QCBOREncodeContext;
+#ifdef CONFIG_ZCBOR
```

Comment:
Is this necessary?` CONFIG_ZCBOR` is a hard requirement for the RPC service, I think it would be cleaner to make sure that `rpc.h` isn't included by any other source files if `CONFIG_ZCBOR` (or more likely `CONFIG_GOLIOTH_RPC`) isn't selected. Alternatively, I'd also be open to making `CONFIG_ZCBOR` selected by `CONFIG_GOLIOTH` instead of by the individual services. I can't imagine anybody using Golioth but not using fw update, rpc, and settings. It's not as direct, but it would simplify things for us.

Reply:
> I'd prefer to keep net/golioth.h minimal, with just "core" code. Things like RPC, LightDB State, LightDB Stream, etc. should go into separate headers.

I agree with this, and I think if we had it that way then we wouldn't need this guard (applications that include `rpc.h` but don't enable `CONFIG_GOLIOTH_RPC` will rightfully fail to compile). Those changes are out of scope for this ticket though, so I'm good with what's here.